tests/lib/date_time/CMakeLists.txt

```diff
@@ -14,7 +14,8 @@ test_runner_generate(src/main.c)
 
 cmock_handle(${ZEPHYR_NRFXLIB_MODULE_DIR}/nrf_modem/include/nrf_modem_at.h
 	     FUNC_EXCLUDE ".*nrf_modem_at_scanf"
-	     FUNC_EXCLUDE ".*nrf_modem_at_printf")
+	     FUNC_EXCLUDE ".*nrf_modem_at_printf"
+	     WORD_EXCLUDE "__nrf_modem_(printf|scanf)_like\(.*\)")
```

Comment:
I don't fully understand why this is needed only in date_time tests and not location, lte_lc, lte_lc_api and sms tests which also has mock for `nrf_modem_at.h` and excludes `nrf_modem_at_printf` and `nrf_modem_at_scanf`.

Reply:
Hmm, looking quickly, seems those other tests are not run by twister so I'm expecting this to cause regression.

Reply:
Alright, good catch.

Reply:
Added